camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/mapred/EtlInputFormat.java

```diff
@@ -263,6 +262,10 @@ public String createTopicRegEx(HashSet<String> topicsSet) {
             // We only care about LeaderNotAvailableCode error on partitionMetadata level
             // Error codes such as ReplicaNotAvailableCode should not stop us.
             if (partitionMetadata.errorCode() == ErrorMapping.LeaderNotAvailableCode()) {
+              partitionMetadata = refreshPartitionMetadata(partitionMetadata, topicMetadata, context);
```

Comment:
I will add a constant for retrying multiple times. But I am not sure if the backoff is necessary here. The method getKafkaMetadata contacts the all the brokers in the re-shuffled broker list for the topicMetadata.